### ISSUE TITLE
Include a configuration-option for Unicode separator characters

### ISSUE THREAD

**Issue by ELLIOTTCABLE** [itchyny/lightline.vim#265]
At the moment, characters that Vim renders as wide can screw up the alignment of Lightline's components:

<img width="533" alt="screen shot 2017-11-24 at 20 28 47" src="https://user-images.githubusercontent.com/200/33227713-2405c604-d156-11e7-8aa7-86317914dd64.png">

The only difference there is the `subseparator`:

```viml
let g:lightline.subseparator = { 'left': "|", 'right': "|" }
let g:lightline.subseparator = { 'left': "\uFE19", 'right': "\uFE19" }
```

Since the latter is displayed as a two-column character by Vim, the spacing looks substantially off. (I see similar problems with some Unicode characters in *other* ranges that are rendered by Vim as *single*-column, but are wide enough to cover part of the following column. That, however, is more easily fixed, by hardcoding additional padding.)

tl;dr I think the padding-whitespace should be *included in the setting* — that would allow for a configuration like the following:

```viml
let g:lightline.subseparator = { 'left': " | ", 'right': " | " }
let g:lightline.subseparator = { 'left': " \uFE19", 'right': " \uFE19" }
```

**Comment by itchyny** [comment]
I cannot change the behavior because it will affect all the statusline of the users. Also, configuring the component type to `raw` will allow the user to trim the spaces.

```vim
let g:lightline.component_type = { 'bufnum': 'raw', 'filename': 'raw', 'filetype': 'raw', 'fileencoding': 'raw' }
```

**Comment by ELLIOTTCABLE** [comment]
Ahhah! Well, that's a sufficient work-around; perhaps a mention of this in the README, then? `raw` isn't really explained anywhere in the documentation.

**Comment by ELLIOTTCABLE** [comment]
Oh, scratch that. I'd have to mark *every single* one of my elements as `raw`, which completely defeats the point of using this plugin.

There's gotta be a better way. Perhaps my suggested change could wait until a major-version bump? Or be a default-off configuration option (`g:lightline.raw_separators = v:true`)?

**Comment by itchyny** [comment]
The option is documented in the help (:h lightline.component_type), and I'm not going to write in README.md, which is a guide for beginners. I don't think another option is necessary here because the option is enough to specify the raw type of the components. Adding additional option requires another codes to check the components are raw or not and leading the plugin to be less *light*.

```vim
let g:lightline.component_type = {}
for s:group in g:lightline.active.left + g:lightline.active.right
  for s:name in s:group
    let g:lightline.component_type[s:name] = 'raw'
  endfor
endfor
unlet! s:name s:group
```

**Comment by ELLIOTTCABLE** [comment]
The problem is, the *component* isn't the thing that's raw (or causing the problems) — it's super-unintuitive to disable the built-in separator, and then manually add a separator to *every single component*, also then making every single component `raw`, *also* then doing all of the math to add any padding … you get the idea.

I dunno. I can accept “this isn't in scope for my project”, so I'll stop commenting, but … you've *gotta* be able to see that that's far from an ideal situation? You're saying a user literally needs to re-implement your entire tool, *inside of* your tool, to use it with double-width Unicode characters?

Oh well.` ¯\_(ツ)_/¯`

**Comment by itchyny** [comment]
I admit that raw type is actually confusing name but I didn't come up with some better naming. Item group `%(...%)` in statusline is useful because it collapse when the content is empty. And I think it looks nice with both the default ascii and powerline-font separators (I use the triangle separators in my config). From my viewpoint, I see very few people complaining about the spacing. I'm sorry but I don't implement features for these people which might cause a performance regression. If I reimplement the next line plugin, I would completely drop configuration for separators.
